consumer/src/solana.rs

```diff
@@ -661,34 +664,61 @@ impl<'a> TransferBackend for CompressedRef<'a> {
+        let merkle_tree_address = Pubkey::from_str(&compression_leaf.merkle_tree)?;
+
+        let merkle_tree_account_data = solana_rpc.get_account_data(&merkle_tree_address)?;
+        let merkle_tree_account =
+            ConcurrentMerkleTreeHeader::try_from_slice(&merkle_tree_account_data)?;
+
+        let asset = asset_api
+            .get_asset(&collection_mint_id)
+            .await
+            .context("fetching asset from DAA")?;
+        let asset_proof = asset_api
+            .get_asset_proof(&collection_mint_id)
             .await
-            .context("Error getting asset data")?;
+            .context("fetching asset proof from DAA")?;
+
+        let root: Vec<u8> = asset_proof.root.into();
+        let data_hash: Vec<u8> = asset.compression.data_hash.context("no data hash")?.into();
+        let creator_hash: Vec<u8> = asset
+            .compression
+            .creator_hash
+            .context("no creator hash")?
+            .into()?;
+        let leaf_id = asset.compression.leaf_id;
+        let proofs = asset_proof
+            .proof
+            .into_iter()
+            .map(|proof| Ok(AccountMeta::new_readonly(proof.try_into()?, false)))
+            .collect::<Result<Vec<AccountMeta>>>()?;
+
+        let mut accounts = vec![
+            AccountMeta::new(tree_authority_address, false),
+            AccountMeta::new_readonly(owner, true),
+            AccountMeta::new_readonly(owner, false),
+            AccountMeta::new_readonly(recipient, false),
+            AccountMeta::new(merkle_tree_address, false),
+            AccountMeta::new_readonly(spl_noop::ID, false),
+            AccountMeta::new_readonly(spl_account_compression::ID, false),
+            AccountMeta::new_readonly(system_program::ID, false),
+        ];
+
+        accounts.extend(proofs);
 
         let instructions = [Instruction {
             program_id: mpl_bubblegum::ID,
-            accounts: [
-                AccountMeta::new(self.0.bubblegum_tree_authority, false),
-                AccountMeta::new_readonly(owner, true),
-                AccountMeta::new_readonly(owner, false),
-                AccountMeta::new_readonly(recipient, false),
-                AccountMeta::new(self.0.bubblegum_merkle_tree, false),
-                AccountMeta::new_readonly(spl_noop::ID, false),
-                AccountMeta::new_readonly(spl_account_compression::ID, false),
-                AccountMeta::new_readonly(system_program::ID, false),
-            ]
-            .into_iter()
-            .collect(),
+            accounts,
             data: mpl_bubblegum::instruction::Transfer {
-                root: todo!("how does DAA work"),
-                data_hash: todo!("how does DAA work"),
-                creator_hash: todo!("how does DAA work"),
-                nonce: todo!("how does DAA work"),
-                index: todo!("how does DAA work"),
+                root: root.try_into()?,
```

Comment:
@ray-kast could you help here. I'm having trouble converting Base58 struct to [u8; 32]